### DISCUSSION TITLE
Download a file using ada?

### DISCUSSION THREAD

**Discussion by flaviu22** [ada-url/ada#572] [Q&A] [answered]
If `ada` is an alternative for `curl`, how can be used to download a file? Does `ada` have `[curl_easy_perform](https://curl.se/libcurl/c/curl_easy_perform.html)` like method? Is there any example of how can `ada` be used to download a file?

**Comment by lemire** [comment, marked as answer]
The ada library is part of the Node.js runtime. If you download a file using the Node.js runtime, it will use ada:

```JavaScript
const request = require('request');
const fs = require('fs');

const downloadFile = (url, filePath) => {
  const file = fs.createWriteStream(filePath);
  request(url).pipe(file).on('close', () => {
    console.log('File downloaded successfully.');
  }).on('error', (err) => {
    fs.unlink(filePath);
    console.error(`Error while downloading file: ${err.message}`);
  });
};

downloadFile('https://www.google.com', 'index.html');
```

> If ada is an alternative for curl,

The ada library is an alternative to the curl component that parses URLs.

If you found some documentation that implied that ada was an alternative to curl as a whole, please point it out to us so we correct it.

**Reply by flaviu22** [reply]
"_The ada library is an alternative to the curl component that parses URLs_"
It's clear to me now, thanks.
